extensions/client/firebase.go

```diff
@@ -0,0 +1,131 @@
+package client
+
+import (
+	"context"
+	"encoding/json"
+	firebase "firebase.google.com/go/v4"
+	"firebase.google.com/go/v4/messaging"
+	"github.com/sirupsen/logrus"
+	"github.com/topfreegames/pusher/interfaces"
+	"google.golang.org/api/option"
+	"time"
+)
+
+type firebaseClientImpl struct {
+	firebase *messaging.Client
+	logger   *logrus.Logger
+}
+
+var _ interfaces.PushClient = &firebaseClientImpl{}
+
+func NewFirebaseClient(jsonCredentials string, logger *logrus.Logger) (interfaces.PushClient, error) {
```

Comment:
Would be better if `ctx` was being received from the function params

Reply:
Done